### PR TITLE
Prevent invalid settings from being saved.

### PR DESCRIPTION
**Pull request by benbowler**
## Summary

Addresses issue:

- #8508

## Relevant technical choices

I went through each module confirming that it's `canSubmitChanges` function correctly checks for changes. Further I created a new `CORE_MODULES` selector, `haveSettingsChanged`, and used this to meet the requirements in the AC, namely:
1. Showing an active "Save" button if nothing has changed
2. Showing an active "Confirm changes" button if the form has changed and is valid
3. Showing and inactive "Confirm changes" button if the form has changed and is invalid

I updated a number of JS tests that had broken as a result of this work, mainly because they were expecting active save buttons without valid changes being made to the field.

I also had to update the E2E tests, removing the `should disconnect the Ads module when removing the Ads Conversion Tracking ID from Ads settings` test, which is no longer possible as the user is not allowed to save an empty string or invalid value. While I was here I fixed a typo in the E2E test names.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Comment by benbowler** [comment]
E2E tests are failing app wide, once #8546 is merged this can be updated from develop and checks re-run.

File: assets/js/googlesitekit/modules/datastore/settings.js [hunk 1]
```diff
@@ -150,6 +150,22 @@ export const selectors = {
...
+	 * @param {string} slug Slug for module store.
+	 * @return {boolean} Whether changes have been made in the settings form.
+	 */
+	haveSettingsChanged: createRegistrySelector(
+		( select ) => ( state, slug ) => {
+			invariant( slug, 'slug is required.' );
+			const storeName = select( CORE_MODULES ).getModuleStoreName( slug );
+			return !! select( storeName )?.haveSettingsChanged?.();
+		}
+	),
```
Comment (tofumatt) [inline assets/js/googlesitekit/modules/datastore/settings.js:168]: We have a generic [`haveSettingsChanged`](https://github.com/google/site-kit-wp/blob/cde25db0ac413e2345df5087c2662b2c86c0e454/assets/js/googlesitekit/data/create-settings-store.js#L288) selector already that takes different arguments, why use the exact same name?

That's a bit confusing for me, maybe this one could be `haveSettingsChangedForModule`, etc.?

Though I'm not sure why we wouldn't use `select( MODULES_ADS ).haveSettingsChanged()` directly, above https://github.com/google/site-kit-wp/pull/8545/files#diff-35a209af9b8af36a7078eba599222127fa6ef4c9bdaa8b38b5dfecfdd685a2ceR66

**Review by tofumatt — CHANGES_REQUESTED** [review]
Largely looks good, just curious about the need for the new selector. If we do need to keep it, let's rename it for clarity. 🙂

File: assets/js/googlesitekit/modules/datastore/settings.js (hunk 1 above)
Comment (benbowler) [reply assets/js/googlesitekit/modules/datastore/settings.js:168]: @tofumatt In terms of the selector naming I was following the naming of the `canSubmitChanges` selector above which shares the naming of the underlaying module selector:

https://github.com/google/site-kit-wp/pull/8545/files#diff-80a2c5f54e6197f4fad0b4cf508c8337f5af754bf8ddd8b6286f65db5b020442R148-R152

The reason we can't use the module level selector is that the `assets/js/components/settings/SettingsActiveModule/Footer.js` component is shared between each module settings, not just the Ads module.

**Review by tofumatt — APPROVED** [review]

**Pull request merged**